src/App.tsx

```diff
@@ -1,24 +1,225 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
+import React, { ChangeEvent, useEffect, useMemo, useState } from "react";
+import { UserWarning } from "./UserWarning";
+import { getTodos, postTodo, deleteTodo, patchTodo } from "./api/todos";
+import { TodoInterface } from "./types/Todo";
+import { TodoList } from "./Components/TodoList/TodoList";
+import { Error } from "./Components/Error/Error";
+import { FilterTodo } from "./Components/FilterTodo/FilterTodo";
+import { FilterStatus } from "./types/FilterStatus";
+import { Header } from "./Components/Header/Header";
+import classNames from "classnames";
 
-const USER_ID = 0;
+const USER_ID = 6429;
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<TodoInterface[]>([]);
+  const [error, setError] = useState("");
+  const [title, setTitle] = useState("");
+  const [temporaryTodo, setTemporaryTodo] = useState<TodoInterface>();
+  const [loadingIds, setLoadingIds] = useState<number[]>([]);
+  const [toggle, setToggle] = useState(false);
+  const [filter, setFilter] = useState<FilterStatus>(FilterStatus.all);
+  const activeTodosСount = useMemo(() => {
+    return todos.filter(({ completed }) => !completed).length;
+  }, [todos]);
+
+  useEffect(() => {
+    if (activeTodosСount === 0) {
+      setToggle(true);
+    } else {
+      setToggle(false);
+    }
+  }, [activeTodosСount]);
+
+  const completedTodos = useMemo(() => {
+    return todos.filter(({ completed }) => completed);
+  }, [todos]);
+
+  const resetError = () => {
+    const timeoutID = setTimeout(() => setError(""), 3000);
+
+    return () => {
+      clearTimeout(timeoutID);
+    };
+  };
```

Comment:
You can use `useEffect` instead of calling this function after `error` changing
```suggestion
  const resetError = () => {
    const timeoutID = setTimeout(() => setError(""), 3000);

    return () => {
      clearTimeout(timeoutID);
    };
  };
  
  useEffect(() => {
    if (error.length > 0) resetError();
  }, [error]);
```